test.js

```diff
@@ -107,7 +107,7 @@ const NewCompWithComponent = CompWithComponent.withComponent(OtherComp)`
 
 
 // Typescript
-const Root = styled<RootProps>('div')`
+const Root = styled.div<RootProps>`
   height: ${(props) => props.label ? 72 : 48}px;
 `
```

Comment:
Will it break the former version?

Why not keep the other test?

Reply:
Good point. Will look into it.

Reply:
@sandervspl simplest way to support both might be to add `(?:<.+>(?=\())?` after the `styled` portion:

```regex
(?:([sS][tT][yY][lL][eE][dD](?:<.+>(?=\())?(?:\.[_$[:alpha:]][_$[:alnum:]]*|\(['"][_$[:alpha:]][_$[:alnum:]]*['"]\)|\([_$[:alpha:]][_$\.[:alnum:]]*(?:\s+as\s+.*?)?\))(?:<.+>)?)|(css|keyframes|injectGlobal))\s*(`)
```

It's possible this is too permissive? It would support something like `` styled<Props>('div')<OtherProps>` `` which … I don't think is valid? But maybe that shouldn't be the concern of the matching expression